presto-hive-common/src/main/java/com/facebook/presto/hive/zorder/ZOrder.java

```diff
@@ -66,7 +71,7 @@ public ZOrder(List<Integer> encodingBits)
         List<Integer> bitLengths = new ArrayList<>(encodingBits);
         for (int dimensionIndex = maxBitLength - 1; dimensionIndex >= 0; dimensionIndex--) {
             for (int bitLengthIndex = 0; bitLengthIndex < bitLengths.size(); bitLengthIndex++) {
-                if (bitLengths.get(bitLengthIndex) >= 0) {
+                if (onlyPositive ? bitLengths.get(bitLengthIndex) > 0 : bitLengths.get(bitLengthIndex) >= 0) {
```

Comment:
If in Line 71, we check onlyPositive value, and adjust values in bitLengths accordingly, then we don't need to check onlyPositive for so many times. E.g.,

if (!onlyPositive) {
   for (int i = 0; i < bitLengths.size(); ++i) {
      bitLengths.set(i, bitLengths.get(i) + 1);
  }
}

Reply:
+1, I'm not sure if JIT compiler would be smart enough to optimize the looping branch

Reply:
This is a good exercise to understand how JIT works for this case. :)